rmw_connextdds_common/src/common/rmw_context.cpp

```diff
@@ -922,6 +968,37 @@ rmw_api_connextdds_init(
   ctx->optimize_large_data = '\0' == disable_optimize_large_data_env[0];
 #endif /* RMW_CONNEXT_DEFAULT_LARGE_DATA_OPTIMIZATIONS */
 
+  /* Lookup and configure initial peer from environment */
+  const char * initial_peers = nullptr;
+  lookup_rc =
+    rcutils_get_env(RMW_CONNEXT_ENV_INITIAL_PEERS, &initial_peers);
```

Comment:
couldn't `NDDS_DISCOVERY_PEERS` be used directly?

Reply:
Yes, it can be, and it probably should be for `rmw_connextdds`. `RMW_CONNEXT_INITIAL_PEERS` was introduced mostly to support `rmw_connextddsmicro` which doesn't have the capability of configuring initial peers from environment itself.

Btw, this is [mentioned in the README section](https://github.com/ros2/rmw_connextdds/blob/d3c3265a92f6294e67376f837ecf8d37f0de7c24/README.md#RMW_CONNEXT_INITIAL_PEERS), inviting Pro users to use `NDDS_DISCOVERY_PEERS`:

> While both `rmw_connextdds` and `rmw_connextddsmicro` will honor this variable,
[equivalent, and more advanced, functionality is already available in RTI Connext DDS](https://community.rti.com/static/documentation/connext-dds/6.0.1/doc/manuals/connext_dds/html_files/RTI_ConnextDDS_CoreLibraries_UsersManual/Content/UsersManual/ConfigPeersListUsed_inDiscov.htm),
for example using variable `NDDS_DISCOVERY_PEERS`.
>
> For this reason, only users of `rmw_connextddsmicro` should consider specifying
`RMW_CONNEXT_INITIAL_PEERS`.

Reply:
> Btw, this is mentioned in the README section, inviting Pro users to use NDDS_DISCOVERY_PEERS:

Ah sorry, I didn't see that when reviewing.

Reply:
No problem, just wanted to point it out for completeness!